app/models/group.rb

```diff
@@ -842,6 +840,32 @@ def update_primary_group
     end
   end
 
+  def self.automatic_membership_users(domains, group_id = nil)
+    domains = domains.gsub('.', '\.')
+
+    users = User.joins(:user_emails).where("user_emails.email ~* '@(#{domains})$'").activated.where(staged: false)
```

Comment:
I don't like how domains is inserted here as it looks like an opportunity for SQL injection. Can you change it to be parameter instead?

(I realize the old code did this too, but this is an opportunity to harden our app.)